ERCS/erc-5559.md

```diff
@@ -1,471 +1,449 @@
 
-In step 1, a standard blockchain call operation is made to the contract. The contract either performs the operation as intended or reverts with an error that specifies the type of handler that the mutation is being deferred to and the corresponding parameters required to perform the subsequent mutation. There are two types of errors that the contract can revert with, but more may be defined in other EIPs:
+```solidity
+// Function in L2 contract
+function setValue(
+    bytes32 node,
+    bytes32 key,
+    bytes32 value
+) external {
+    // Some code storing data mapped by node & msg.sender
+    ...
+}
+```
 
-- `StorageHandledByL2(chainId, contractAddress)`
-- `StorageHandledByOffChainDatabase(sender, url, data)`
+### Database Handler: `StorageHandledByDatabase()`
+A minimal database handler is similar to an L2 in the sense that:
 
-In step 2, the client builds and performs a new request based off of the type of error received in (1). These handshakes are outlined in the sections below:
+  a) Similar to `chainId`, it requires the `gatewayUrl` that is tasked with handling off-chain write operations, and
 
-- [StorageHandledByL2](#data-stored-in-an-l2)
-- [StorageHandledByOffChainDatabase](#data-stored-in-an-off-chain-database) 
+  b) Similar to `eth_call`, it requires `eth_sign` output to secure the data, and the client must prompt the users for these signatures.
 
-In some cases, the mutation may be deferred multiple times
-- [Storage Deferred Twice L1 > L2 > Off-Chain](#data-stored-in-an-l2--an-off-chain-database) 
+This specification does not require any other data to be stored on L1 other than the bespoke `gatewayUrl`; the storage handler therefore should only return the `gatewayUrl` in revert.
 
-### Data Stored in an L1
-```
-┌──────┐                ┌───────────┐ 
-│Client│                │L1 Contract│ 
-└──┬───┘                └─────┬─────┘ 
-   │                          │       
-   │ somefunc(...)            │       
-   ├─────────────────────────►│       
-   │                          │       
-   │ response                 │       
-   │◄─────────────────────────┤       
-   │                          │       
+```solidity
+error StorageHandledByDatabase(
+    string gatewayUrl
+);
+
+// Generic function in a contract
+function setValue(
+    bytes32 node,
+    bytes32 key,
+    bytes32 value
+) external {
+    (
+        string gatewayUrl // Gateway URL; may be globally constant
+    ) = getMetadata(node);
+    // gatewayUrl = "https://api.namesys.xyz"
+    // Defer write call to database handler
+    revert StorageHandledByDatabase( 
+        gatewayUrl
+    );
+};
 ```
 
-In the case in which no reversion occurs, data is stored in the L1 contract when the transaction is executed.
+Following the revert, the client must take these steps:
 
-### Data Stored in an L2
```

Comment:
I would suggest that you keep `Data Stored in an L1` and `Data Stored in an L2` because there are no changes

Reply:
The language in this document is completely different. By using your suggestions, the entire document will read as if some sections were written by a completely different individual than the rest of the document. It destroys coherence.

Reply:
> Data Stored in an L1

I will add this back in the next commit

Reply:
> > Data Stored in an L1
> 
> I will add this back in the next commit

Fixed in https://github.com/ethereum/ERCs/pull/388/commits/eb07bce4d395e1fcefefa3d3b77e06ecb3f22733